src/charm.py

```diff
@@ -459,20 +459,21 @@ def _set_alerts(self, container) -> bool:
                 alert rule files need to be created. This container
                 must be in a pebble ready state.
 
-        Returns: A boolean indicating if new alert rules were pushed.
+        Returns: A boolean indicating if new or different alert rules were pushed.
         """
+        alert_rules_changed = True
         metrics_consumer_alerts = self.metrics_consumer.alerts()
         remote_write_alerts = self.remote_write_provider.alerts()
-
         alerts_hash = sha256(str(metrics_consumer_alerts) + str(remote_write_alerts))
+
         if alerts_hash == self._stored.alerts_hash:
-            return False
+            alert_rules_changed = False
```

Comment:
Seems like the logic around `alert_rules_changed` could be replaced with:

```python
alert_rules_changed = alerts_hash != self._stored.alerts_hash
```

Reply:
do you mean something like this?

```python
    def _set_alerts(self, container) -> bool:
        """Create alert rule files for all Prometheus consumers.

        Args:
            container: the Prometheus workload container into which
                alert rule files need to be created. This container
                must be in a pebble ready state.

        Returns: A boolean indicating if new or different alert rules were pushed.
        """
        metrics_consumer_alerts = self.metrics_consumer.alerts()
        remote_write_alerts = self.remote_write_provider.alerts()
        alerts_hash = sha256(str(metrics_consumer_alerts) + str(remote_write_alerts))

        container.remove_path(RULES_DIR, recursive=True)
        self._push_alert_rules(container, self.metrics_consumer.alerts())
        self._push_alert_rules(container, self.remote_write_provider.alerts())
        self._stored.alerts_hash = alerts_hash
        return alerts_hash != self._stored.alerts_hash
```

Reply:
Not quite: at the return statement they would always match because the stored value was updated one line above.